web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php

```diff
@@ -543,41 +536,31 @@ public function work(string $wid): array {
    *   An array of base urls.
    */
   public static function externalApiBaseUrls(): array {
-    /** @var \Drupal\dpl_library_agency\GeneralSettings $general_settings */
-    $general_settings = \Drupal::service('dpl_library_agency.general_settings');
     $react_apps_settings = \Drupal::configFactory()->get('dpl_react_apps.settings');
     $fbs_settings = \Drupal::config(FbsSettingsForm::CONFIG_KEY);
 
     /** @var \Drupal\dpl_publizon\DplPublizonSettings $publizon_settings*/
     $publizon_settings = \Drupal::service('dpl_publizon.settings');
 
     // Get base urls from this module.
-    $services = $react_apps_settings->get('services') ?? [];
-
-    // The base url of the FBI service is a special case
-    // because a part (the profile) of the base url can differ.
-    // Lets' handle that:
-    if (!empty($services) && !empty($services['fbi']['base_url'])) {
-      $placeholder_url = $services['fbi']['base_url'];
-      foreach ($general_settings->getFbiProfiles() as $type => $profile) {
-        $service_key = sprintf('fbi-%s', $type);
-        // The default FBI service has its own key with no suffix.
-        if ($type === FbiProfileType::DEFAULT->value) {
-          $service_key = 'fbi';
-        }
-        // Create a service url with the profile embedded.
-        $base_url = preg_replace('/\[profile\]/', $profile, $placeholder_url);
-        $services[$service_key] = ['base_url' => $base_url];
+    $configuredServices = $react_apps_settings->get('services') ?? [];
+    $services = \Drupal::moduleHandler()->invokeAll('dpl_react_apps_api_urls');
+    foreach ($configuredServices as $name => $conf) {
+      // @todo remove when the FBI URL configuration is moved to dpl_fbi.
+      if ($name == 'fbi') {
+        continue;
       }
+
+      $services[$name] = $conf['base_url'];
     }
```

Comment:
Please consider if this code is necessary.

As I see it FBI url configuration has now been moved to `dpl_fpi`.